### DISCUSSION TITLE
clipboard.js

### DISCUSSION THREAD

**Discussion by vulpes2** [thegeeklab/hugo-geekblog#52] [Ideas]
Hello, I've been testing this theme and I noticed clipboard.js was being unconditionally loaded on every single page. Ideally I would like to avoid downloading it (and clipboard-loader) at all on pages that doesn't have the "copy to clipboard" buttons to minimize js usage, is this something feasible to implement?

**Comment by xoxys** [comment]
Hi, I don't really see how this should work, as it depends on the content added to a page whether the library is needed or not. Also, clipboard.js + loader has a size of ~12 kb, so I would call it a micro-optimization. For example, any default font we use for this theme is 10x of that size.  So optimizing the loading of clipboard.js doesn't seem like the right way to me, sorry.

**Comment by vulpes2** [comment]
Sure thing, thanks for the explanation.
